api/handlers/dataexplorerhandler.py

```diff
@@ -251,6 +251,91 @@ def _parse_request(self, request_type='search'):
 
         return return_type, modified_filters, search_string
 
+    @require_login
+    def get_type_ahead_values(self):
+        """
+        Return list of type ahead values for a key given a value 
+        that the user has already started to type in for the value of 
+        a custom string field
+        """
+        user_value = self.request.json_body['value']
+        custom_field = self.request.json_body['field_name']
+        modified_filters = []
+        # Add permissions filter to list if user is not requesting all data
+        if not self.request.json_body.get('all_data', False):
+            modified_filters.append({'term': {'permissions._id': self.uid}})
+
+        body = {
+            "size": 0, 
+            "query": {
+                "bool": {
+                    "must" : {
+                        "match" : { custom_field : user_value}
+                    },
+                    "filter" : {
+                        "bool" : {
+                            "must" : modified_filters
```

Comment:
If the user permission is not added to the filter, we wouldn't want an empty `must` clause.